sources/HUBComponentModelBuilderImplementation.m

```diff
@@ -550,9 +594,18 @@ - (HUBComponentModelBuilderImplementation *)getOrCreateBuilderForChildWithIdenti
                                                                                                                       iconImageResolver:self.iconImageResolver
                                                                                                                    mainImageDataBuilder:nil
                                                                                                              backgroundImageDataBuilder:nil];
+    newBuilder.delegate = self;
     
     self.childBuilders[newBuilder.modelIdentifier] = newBuilder;
     [self.childIdentifierOrder addObject:newBuilder.modelIdentifier];
+    if (groupIdentifier) {
+        NSString *nonNilGroupIdentifier = groupIdentifier;
+        if (!self.childBuildersByGroupIdentifier[nonNilGroupIdentifier]) {
```

Comment:
Do you need this? The delegate will be called whenever the group identifier is changed anyway, which will add the builder to the right group.